netmiko/cisco/cisco_bsp.py

```diff
@@ -0,0 +1,284 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+from logger.cafylog import CafyLog
+from netmiko.cisco_base_connection import CiscoBaseConnection
+from netmiko.ssh_exception import NetMikoAuthenticationException
+import re
+import time
+
+log = CafyLog()
+
+class CiscoBsp(CiscoBaseConnection):
+    '''
+    CiscoBsp is based of CiscoBaseConnection
+    '''
+    def bmc_to_bsp_prompt(self,bmc_prompt_pattern=r'\s*\#',bsp_prompt_pattern=r']\s*\#',delay_factor=1,max_loops=20):
+        '''
+        Switch from BMC to BSP prompt
+
+        :param bmc_prompt_pattern: expected BMC prompt
+        :param bsp_prompt_pattern: expected BSP prompt
+        :param delay_factor: Factor to adjust delays
+        :param max_loops: max number of iterations for attempting to switch prompts
+        :return:
+        '''
+
+        self.TELNET_RETURN = '\n'
+        bmc_to_bsp_cmd = "sol.sh"
+        delay_factor = self.select_delay_factor(delay_factor)
+
+        i = 1
+        while i <= max_loops:
+            i += 1
+            output = self.find_prompt()
+            if re.search(bmc_prompt_pattern, output):
+                log.debug("On BMC Prompt")
+                log.debug("Sol.sh to enter BSP prompt")
+                self.write_channel(bmc_to_bsp_cmd)
+                time.sleep(1 * delay_factor)
+                self.write_channel(self.TELNET_RETURN)
+                time.sleep(1 * delay_factor)
+                output = self.find_prompt()
+
+            if re.search(bsp_prompt_pattern, output):
```

Comment:
this isn't a full check. the sol.sh is calling 3 or more .sh script to turn the termserver in linux for bsp and output text.
if that fails , this is not gonna move the prompt and line 228 is nothing and whle operations after that will fail.
ok?

Reply:
if bmc to bsp is successful then we are returning back from while loop.
raising exception after while loop ends for max_loops number of iterations.